### PR TITLE
packet: support creating a builder from an existing packet

### PR DESCRIPTION
**Pull request by ystreet**

### CODE REVIEW

File: src/builder.rs [hunk 1]
```diff
@@ -73,12 +73,24 @@ impl<'a> RtpPacketBuilder<'a> {
         self
     }
 
+    /// Set 1. whether padding is used and 2. the number of padding bytes to use for this packet
+    pub fn maybe_padding(mut self, padding: Option<u8>) -> Self {
```
Comment (sdroege) [inline src/builder.rs:75]: `padding` can be at most 3 or not? Is there a check for that somewhere? Also why is this needed, wouldn't padding be added automatically as needed when creating the actual packet?

File: src/packet.rs [hunk 2]
```diff
@@ -251,6 +250,28 @@ impl<'a> RtpPacket<'a> {
         };
         &self.data[offset..self.data.len() - pad]
     }
+
+    /// Creates a builder that will be able to reconstruct this packet byte for byte (exluding
```
Comment (sdroege) [inline src/packet.rs]:
```suggestion
    /// Creates a builder that will be able to reconstruct this packet byte for byte (excluding
```

File: src/packet.rs [hunk 3]
```diff
@@ -251,6 +250,28 @@ impl<'a> RtpPacket<'a> {
...
+    /// any padding bytes).  One can modify any aspect of the returned builder to fit your needs.
```
Comment (sdroege) [inline src/packet.rs]:
```suggestion
    /// any padding bytes).  Any aspect of the returned builder can be modified.
```

**Review by sdroege — APPROVED** [review]

File: src/builder.rs (hunk 1 above)
Comment (ystreet) [reply src/builder.rs:75]: Padding can be used to align to a 4-byte boundary, it can also be used to align to an encryption cipher's block size which may be larger than 4. As such, we don't limit the actual size of the padding.

Comment (ystreet) [reply src/builder.rs:75]: Also, padding is not automatically added as the size of a RTP packet does not have to align to a 4 byte boundary.  It is delimited by the size of the UDP datagram (or some other external size).

Comment (sdroege) [reply src/builder.rs:75]: Ah, good to know

**Review by sdroege — APPROVED** [review]
